src/jetstream.rs

```diff
@@ -731,33 +664,692 @@ impl JetStream {
+            if wants_flow_control {
+                return Err(io::Error::new(
+                    io::ErrorKind::InvalidInput,
+                    "queue subscription doesn't support flow control",
+                ));
+            }
+        };
+
+        // Checks specific to ordered consumers.
+        if let Some(options) = maybe_options.filter(|options| options.ordered) {
+            // Check for queue subscription.
+            if maybe_queue.is_some() {
+                return Err(io::Error::new(
+                    io::ErrorKind::Other,
+                    "queues not be set for an ordered consumer",
+                ));
+            }
+
+            // Check for durable name
+            if options.durable_name.is_some() {
+                return Err(io::Error::new(
+                    io::ErrorKind::Other,
+                    "durable name can not be set for an ordered consumer",
+                ));
+            }
+
+            // Check for bound consumers.
+            if options.consumer_name.is_some() {
+                return Err(io::Error::new(
+                    io::ErrorKind::Other,
+                    "can not bind existing consumer for an ordered consumer",
+                ));
+            }
+
+            // Check for ack policy.
+            if options.ack_policy.is_some() {
+                return Err(io::Error::new(
+                    io::ErrorKind::Other,
+                    "ack policy can not be set for an ordered consumer",
+                ));
+            }
+
+            // Check for max deliver.
+            if options.max_deliver.is_some() {
+                return Err(io::Error::new(
+                    io::ErrorKind::Other,
+                    "max deliver can not be set for an ordered consumer",
+                ));
+            }
+
+            // Check for deliver subject (we pick our own).
+            if options.deliver_subject.is_some() {
+                return Err(io::Error::new(
+                    io::ErrorKind::Other,
+                    "deliver subject can not be set for an ordered consumer",
+                ));
+            }
+        }
+
+        let process_consumer_info = |info: ConsumerInfo| {
+            // Make sure this new subject matches or is a subset.
+            if !info.config.filter_subject.is_empty() && subject != info.config.filter_subject {
+                return Err(io::Error::new(
+                    io::ErrorKind::Other,
+                    "subject does not match consumer",
+                ));
+            }
+
+            if let Some(deliver_group) = info.config.deliver_group.as_ref() {
+                if let Some(queue) = maybe_queue {
+                    if deliver_group != queue {
+                        return Err(io::Error::new(
+                            io::ErrorKind::Other,
+                            format!(
+                                "cannot create a queue subscription {} for a consumer with a deliver group {}",
```

Comment:
I would prefer error telling user why he cannot create the sub  "cannot create a queue subscription if deliver name is set and does not match queue name"

Reply:
No opinion either way, error messages are taken from the Go client.

Reply:
We should include the error from the server at minimum.

Reply:
Server errors are always reported, this is client side validation.